src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/Reindex/ReindexJobCosmosThrottleController.cs

```diff
@@ -0,0 +1,113 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Core;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.Reindex;
+using Microsoft.Health.Fhir.Core.Features.Operations.Reindex.Models;
+using Microsoft.Health.Fhir.CosmosDb.Features.Metrics;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Operations.Reindex
+{
+    public class ReindexJobCosmosThrottleController : IReindexJobThrottleController, INotificationHandler<CosmosStorageRequestMetricsNotification>
+    {
+        private int? _configuredRUThroughput;
+        private DateTimeOffset? _intervalStart = null;
+        private double _rUsConsumedDuringInterval = 0.0;
+        private ushort _delayFactor = 0;
+        private double? _targetRUs = null;
+        private bool _initialized = false;
+
+        public ReindexJobCosmosThrottleController(int? configuredRUThroughput)
+        {
+            _configuredRUThroughput = configuredRUThroughput;
```

Comment:
Does this work if a database has been manually changed?

Reply:
Update to read the provisioned RUs at runtime.